### PR TITLE
create GET /api/articles/:article_id/comments endpoint

### PR DESCRIPTION
**Pull request by ViktoriiaPavliukh**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -122,4 +122,43 @@ describe("app", () => {
         });
     });
   });
+  describe("GET /api/articles/:article_id/comments", () => {
```
Comment (aloush-dev) [inline __tests__/api.test.js:125]: I'd like to see a test for what happens if you get an empty array back as there is no comments.
This would still be aa 200 success.

File: models/article.models.js [hunk 2]
```diff
@@ -36,3 +36,29 @@ exports.selectArticleById = (article_id) => {
...
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;",
+      [article_id]
+    )
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `No comments found for article_id: ${article_id}`,
+        });
+      }
```
Comment (aloush-dev) [inline models/article.models.js]: As I mentioned within the test file the comments coming back as empty would not be an error.
But please add a test to show that you do infact get an empty array returned.

File: models/article.models.js [hunk 3]
```diff
@@ -36,3 +36,29 @@ exports.selectArticleById = (article_id) => {
...
+      return result.rows;
+    })
+    .catch((err) => {
+      if (err.code === "22P02") {
+        return Promise.reject({
+          status: 400,
+          msg: "Invalid article_id",
+        });
+      }
+      throw err;
```
Comment (aloush-dev) [inline models/article.models.js]: This logic should be handeled in your app.js with error handlers.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Please see and address comments and resubmit.

File: models/article.models.js [hunk 4]
```diff
@@ -27,12 +27,6 @@ exports.selectArticleById = (article_id) => {
       return result.rows[0];
     })
     .catch((err) => {
-      if (err.code === "22P02") {
-        return Promise.reject({
-          status: 400,
-          msg: "Invalid id",
-        });
-      }
       throw err;
```
Comment (nc-poppy-m) [inline models/article.models.js]: You're right to remove the promise.reject in the catch block, but we also don't want to throw an error here. All we want is to pass it to the _next_ error handling middleware. How might you do that?

File: models/article.models.js [hunk 5]
```diff
@@ -45,20 +39,11 @@ exports.selectCommentsByArticleId = (article_id) => {
     )
     .then((result) => {
       if (result.rows.length === 0) {
```
Comment (nc-poppy-m) [inline models/article.models.js]: If rows is empty, it will come back as an empty array when you return result.rows anyway, so you don't need to repeat this functionality with this ```if``` statement

File: models/article.models.js [hunk 6]
```diff
@@ -45,20 +39,11 @@ exports.selectCommentsByArticleId = (article_id) => {
...
       return result.rows;
     })
     .catch((err) => {
-      if (err.code === "22P02") {
-        return Promise.reject({
-          status: 400,
-          msg: "Invalid article_id",
-        });
-      }
       throw err;
```
Comment (nc-poppy-m) [inline models/article.models.js]: We don't need the throw err here, see above 🙂

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Hey Viktoriia! Really good code so far on the testing side of things, just a few things to note with your error handling, so try and action my comments and resubmit the pr 🙂

File: __tests__/api.test.js [hunk 7]
```diff
@@ -122,4 +122,41 @@ describe("app", () => {
...
+
+    test("400: responds with an error message when given an invalid article_id", () => {
+      return request(app)
+        .get("/api/articles/not-an-article/comments")
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toBe("Invalid id");
+        });
+    });
+  });
```
Comment (Rayhaan2k9) [inline __tests__/api.test.js:171]: You are missing a 404 test here. What if the article doesn't exist? For example : ```/api/articles/9999/comments```

File: models/article.models.js [hunk 8]
```diff
@@ -26,13 +26,23 @@ exports.selectArticleById = (article_id) => {
...
+    .query(
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;",
+      [article_id]
+    )
+    .then((result) => {
+      return result.rows;
+    })
+};
+
+
```
Comment (Rayhaan2k9) [inline models/article.models.js:42]: You will need to add some logic in here to handle the situation of an article that doesn't exist. This may have an effect on the article that does exist but has no comments. Have a think about how you can check the article exists, and if so, ensure it returns an empty array if has no comments

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
Almost there with this one! Just need to add 1 more tests and logic to pass it🙂

**Review by aloush-dev — APPROVED** [review]
Great job implementing feedback!



Merge away.
